compiler/luci/pass/src/SubstituteTransposeToReshapePass.cpp

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/SubstituteTransposeToReshapePass.h"
+
+#include <luci/IR/CircleNodes.h>
+
+namespace
+{
+
+bool substitute_transpose_to_reshape(luci::CircleNode *node)
+{
+  auto target_node = dynamic_cast<luci::CircleTranspose *>(node);
+  if (target_node == nullptr)
+    return false;
+
+  auto perm_const = dynamic_cast<luci::CircleConst *>(target_node->perm());
+  if (perm_const == nullptr)
+    return false;
+
+  assert(perm_const->dtype() == loco::DataType::S32);
+
+  auto input_node = loco::must_cast<luci::CircleNode *>(target_node->a());
+  if (perm_const->dim(0).value() != input_node->rank())
+    return false;
+
+  uint32_t idx = 0;
+  for (uint32_t i = 0; i < perm_const->size<loco::DataType::S32>(); i++)
+  {
```

Comment:
It would be better to add assert to check the value of perm  (e.g., perm_value >= 0 && perm_value < input_node->rank()).

Reply:
Okay I will add assert before value check.